shell.nix

```diff
@@ -60,6 +58,10 @@ mkShellMold {
     pkgs.jq # used in ./tests and ./tools
     pkgs.nixfmt-rfc-style # to format this file
     pkgs.npins # used in tools/nix/update.py
+    pkgs.iconv
+    pkgs.ccache # compile cache for c/cxx code
+    pkgs.sccache # compile cache for rust code
```

Comment:
I don't think we want this by default. Any specific reason for setting it up here?

Reply:
It's already used in a shell hook below unconditionally for `RUST_WRAPPER`. 
I can either install nix package on darwin only, remove ccache/sccache usage from shell hook or make it conditional too. What would you prefer?

Reply:
It was there before, and I prefer not to remove it.

The point of `shell.nix` is not just to have a bare minimum to build Qdrant, but rather to make local development comfortable. So it's a bit opinionated and includes stuff like rust-analyzer, build caching tricks, and stuff to make out git pre-push hooks work.